File: engine/lua/world/state.lua
-- world/state.lua
-- Global state flags.
-- State.set(key, value), State.get(key), State.reset()

local State = {}
local flags = {}

function State.set(key, value) flags[key] = value end
function State.get(key)        return flags[key]   end
function State.reset()         flags = {}          end

return State

File: engine/typescript/src/world/state.ts
// world/state.ts
// Global state flags.
// State.set(key, value), State.get(key), State.reset()

const flags: Record<string, unknown> = {}

export const State = {
    set(key: string, value: unknown): void {
        flags[key] = value
    },
    get(key: string): unknown {
        return flags[key]
    },
    reset(): void {
        for (const k of Object.keys(flags)) delete flags[k]
    },
}
